src/gl.rs

```diff
@@ -529,9 +529,9 @@ pub fn uniform_4f(location: GLint, x: GLfloat, y: GLfloat, z: GLfloat, w: GLfloa
 pub fn uniform_matrix_4fv(location: GLint, transpose: bool, value: &[f32]) {
     unsafe {
         ffi::UniformMatrix4fv(location,
-                               1 as GLsizei,
-                               transpose as GLboolean,
-                               mem::transmute(&value[0]));
+                              (value.len() / 16) as GLsizei,
+                              transpose as GLboolean,
+                              mem::transmute(&value[0]));
```

Comment:
While you’re at it, please replace this line with `value.as_ptr()`. I prefer to avoid `transmute` without type annotations when type inference relies on non-local definitions (here, the parameter of `UniformMatrix4fv`, as opposed to e.g. `transmute::<&f32, *const f32>(…)`). But it’s even better to avoid transmute entirely when possible.